Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -248,6 +252,10 @@ open class RouteController: NSObject {
                   return
         }
         
+        if let route = indexedRouteResponse.currentRoute {
+            alternativesStore?.mainRoute = route
```

Comment:
`mainRoute` here is a starting point to evaluate alternatives. It is safer to setup it before doing a `setRoute` because `setRoute` will trigger detected alternatives emit. If we are setting main route in the callback, there will be a place for race condition between the two